text/0730-semver-for-ts.md

```diff
@@ -0,0 +1,912 @@
+[emi-false]: https://www.typescriptlang.org/play?esModuleInterop=false&target=7#code/CYUwxgNghgTiAEBbA9sArhBByAzsxIAtGgA44AucUihEAlgEYywCeW8A3gLABQ8-8MMgB2FeADNkyAFzwAFAEp4AXgB88CjDrCA5gG5eA+CAAeJZDHIqJUgzwC+vXqEiwEKdJnhZELWo2YYNk5DATpEc0sbZAkYfG88AmIyShBqfyZWLDtQ-lNIqyFRK18AFQALbR1Zbj4jAUkZeHIWEhBkcWi7eoEGWFlFFXVhNEQGEBhugUcHIA
+
+Accordingly, library authors should set both `allowSyntheticDefaultImports` and `esModuleInterop` to `false`. This allows consumers to opt into these semantics, but does not *require* them to do so. Consumers can always safely use alternative import syntaxes (including falling back to `require()` and `import()`), or can enable these flags and opt into this behavior themselves.
+
+(If the Node ecosystem migrates full to ES modules over the next few years, this problem will be substantially mitigated.)
+
+
+### Design summary
+
+This section is a non-normative short summary for easy digestion. See the previous sections and [**Conformance**](#conformance) below for normative text.
+
+-   Public published types are part of the SemVer contract of a package, and must be versioned accordingly, per the specification above.
+
+-   Adding a new TypeScript version to the support matrix *may* cause breaking changes. When it does not, adding it is a normal minor release. When it *does*, cause a breaking change, the package must either mitigated that breakage (so consumers are not broken) *or* the package must release a major version.
+
+-   Removing a TypeScript version from the support matrix is a breaking change, except when it falls out of the supported version range under the “rolling support windows” policy.
+
+-   Packages using the “rolling window” policy should normally support all TypeScript versions released during the current ‘LTS’ of other core packages/runtimes/etc. they support, and drop support for them only when they drop support for that ‘LTS’, to minimize the number of major versions in the ecosystem.
+
+-   Both the currently-supported compiler versions and the compiler version support policy must be documented.
+
+-   Packages must be authored with the following compiler options:
+    -   `strict: true`
+    -   `noPropertyAccessFromIndexSignature: true`
+
+-   Libraries should generally be authored with the following compiler options:
+    -   `esModuleInterop: false`
+    -   `allowSyntheticDefaultImports: false`
+
+
+### Conformance
+
+To conform to this standard, a package must:
+
+- link to the final published version of this specification
+- specify the compiler support policy
+- specify the currently-supported versions of TypeScript
+- specify the definition of “public API” used by the library (e.g. “only documented types” vs. “all published types” etc.)
```

Comment:
I've folded that discussion *into* the definition of public API, per your comments on that section of the definitions, so I think it's fine to leave it as a high-level bullet here – though perhaps it's worth including "user-constructibility" and "subclassing" as examples here since they're important?